Map Entry equality

emidln opened the pull request:
This extends -eq on MapEntry to include vectors (as in clojure) as well as extends -seq to MapEntry so that the reverse -eq (vector to MapEntry) can also be true.

Now the following holds,

```clj
(= (map-entry :a 1) [:a 1])
(= [:a 1] (map-entry :a 1))
```

thomasmulvaney commented:
This looks good however whilst `(= '(:a 1) (map-entry :a 1))` is true. `(= (map-entry :a 1) '(:a 1))` is false.

emidln commented:
Rather than `(cond .... :else false)`, should MapEntry have `(cond .... :else (= (seq self) other))` such that we just defer to seq equality?

emidln commented:
I added the test for (= (map-entry :a 1) '(:a 1)) and added a fallback to calling seq on the map entry and then using = on that and other. This is perhaps not ideal, but until we can dispatch on both types for the -eq protocol, I think it's the best that can be done.

halgari commented:
I'm okay with merging this, but we have a merge conflict, if we can get that fixed, I'll push the button.

emidln commented:
I rebased against master to fix the merge conflict. The tests fail, but they're not tests for *this* code (and they fail in master too).

heyLu commented:
Merged, thanks!
